x/stakeibc/keeper/lsm.go

```diff
@@ -1,12 +1,55 @@
 package keeper
 
 import (
+	"time"
+
+	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	"github.com/golang/protobuf/proto" //nolint:staticcheck
+
 	"github.com/Stride-Labs/stride/v8/x/stakeibc/types"
 )
 
-func (k Keeper) TransferLSMTokenToHost(ctx sdk.Context, lsmTokenDeposit types.LSMTokenDeposit) error {
```

Comment:
What happened to this function?  Where is the logic which checks for TRANSFER_IN_PROGRESS or TRANFER_FAILED states?  Is that just part of a different PR and not relevant here?

Reply:
Sorry I think I confused myself a bit between the spec and the ticket definitions (didn't realize this was assigned to you). 

Before I thought we would have separate Mint and Transfer functions but then I realized the mint code is actually pretty straight forward so I thought i made more sense to just bundle those together into one function.

I wound up putting this code in the `FinishLSMLiquidStake` function [[here](https://github.com/Stride-Labs/stride/pull/713/files#diff-e5e494420dbb9b9ea933fbe70b5bb22b5af4c72a53940963a375dc32f8c754bfR195-R228)]. 

You're still tasked with the callback though!